src/main/java/com/facebook/openwifirrm/ucentral/UCentralUtils.java

```diff
@@ -183,9 +183,23 @@ public static boolean setRadioConfigField(
 				continue;
 			}
 
-			// Compare vs. existing value
-			int currentValue = radioConfig.get(fieldName).getAsInt();
-			if (currentValue == newValue) {
+			// Compare vs. existing value.
+			// not all values are int so override those values
+			Integer currentValue = null;
+			JsonElement fieldValue = radioConfig.get(fieldName);
+			try {
+				currentValue = fieldValue.getAsInt();
+			} catch (NumberFormatException e) {
```

Comment:
Instead of doing this via exception handling you could write:
```
JsonElement e;
if (e.isJsonPrimitive() && e.getAsJsonPrimitive().isNumber()) {
  e.getAsInt();
}
```
These JSON libraries don't make it easy... but it's possible.

Reply:
Yeah I thought about it but then I went back to this but I have no real preference so let me do that